### PR TITLE
feat: disentangle UI from shadowgraph

### LINKED ISSUE #12255: Disentangle UI from shadowgraph
Disentangle the local test/debug UI framework from the shadowgraph. This will be done by exposing a stream of data that the UI can use to reconstruct the data it currently needs.

### PR DESCRIPTION
**Pull request by cody-littley**
closes #12255

### CODE REVIEW

**Review by jeromy-cannon — DISMISSED** [review]

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/publisher/DefaultPlatformPublisher.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * This component is responsible for publishing internal platform data to external subscribers. By default this is not
+ * enabled, and will only publish data if handler methods are registered with the platform at startup time.
+ */
+public class DefaultPlatformPublisher implements PlatformPublisher {
+
+    private final Logger logger = LogManager.getLogger(DefaultPlatformPublisher.class);
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/publisher/DefaultPlatformPublisher.java:35]: `static`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiEventStorage.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    // A note on concurrency: although all input to this class is sequential and thread safe, access to this class
+    // happens asynchronously. This requires all methods to be synchronized.
+
+    private long maxGeneration = FIRST_GENERATION;
+
+    private final Consensus consensus;
+    private final SimpleLinker linker;
+    private final Configuration configuration;
+
+    public GuiEventStorage(@NonNull final Configuration configuration, @NonNull final AddressBook addressBook) {
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiEventStorage.java:55]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 3]
```diff
@@ -297,16 +299,20 @@ public class SwirldsPlatform implements Platform {
...
+     * @param recycleBin                used to delete files that may be useful for later debugging
+     * @param id                        the ID for this node
+     * @param mainClassName             the name of the app class inheriting from SwirldMain
+     * @param swirldName                the name of the swirld being run
+     * @param appVersion                the current version of the running application
+     * @param initialState              the initial state of the platform
+     * @param emergencyRecoveryManager  used in emergency recovery.
+     * @param preconsensusEventConsumer the consumer for preconsensus events, null if publishing these events has not
+     *                                  been enabled
+     * @param snapshotOverrideConsumer  the consumer for snapshot overrides, null if publishing these events has not
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:314]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiEventStorage.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    /**
+     * Handle a preconsensus event. Called after events are released from the orphan buffer.
+     *
+     * @param event the event to handle
+     */
+    public synchronized void handlePreconsensusEvent(@NonNull final GossipEvent event) {
+        maxGeneration = Math.max(maxGeneration, event.getGeneration());
+
+        final EventImpl eventImpl = linker.linkEvent(event);
+        final List<ConsensusRound> rounds = consensus.addEvent(eventImpl);
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/GuiEventStorage.java:83]: Good catch. Fixed.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/SimpleLinker.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    private long nonAncientThreshold = 0;
+
+    /**
+     * Constructor
+     *
+     * @param ancientMode the ancient mode
+     */
+    public SimpleLinker(@NonNull final AncientMode ancientMode) {
+        if (ancientMode == AncientMode.BIRTH_ROUND_THRESHOLD) {
+            this.parentDescriptorMap =
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/SimpleLinker.java:72]:
```
throw new UnsupportedOperationException("not yet supported");
```

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/hashgraph/internal/StandardGuiSource.java [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+import com.swirlds.platform.gui.GuiEventStorage;
+import com.swirlds.platform.gui.hashgraph.HashgraphGuiSource;
+import com.swirlds.platform.internal.EventImpl;
+import com.swirlds.platform.system.address.AddressBook;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A {@link HashgraphGuiSource} that retrieves events from a {@link Shadowgraph}
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/hashgraph/internal/StandardGuiSource.java:29]: removed reference to shadowgraph

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/hashgraph/internal/StandardGuiSource.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    public long getMaxGeneration() {
+        return eventStorage.getMaxGeneration();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    @NonNull
+    public List<EventImpl> getEvents(long startGeneration, int numGenerations) {
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gui/hashgraph/internal/StandardGuiSource.java:60]: added

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/publisher/DefaultPlatformPublisher.java (hunk 1 above)
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/publisher/DefaultPlatformPublisher.java:35]: fixed

File: platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/preconsensus/PcesBirthRoundMigrationTests.java [hunk 8]
```diff
@@ -136,8 +135,11 @@ private PcesFilesWritten generateLegacyPcesFiles(
         final int eventsPerFile = eventCount / fileCount;
         final Instant startingTime = randomInstant(random);
 
+        final PlatformContext platformContext =
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-unit-tests/core/swirlds-platform-test/src/test/java/com/swirlds/platform/test/event/preconsensus/PcesBirthRoundMigrationTests.java:138]: Yup. Not a fan of static platform context even in testing, which is why I didn't revert my change when resolving the merge conflict.

**Review by alittley — APPROVED** [review]

**Review by alittley — DISMISSED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by alittley — DISMISSED** [review]

**Review by alittley — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Pull request merged**